src/lib/message.ts

```diff
@@ -30,6 +30,8 @@ class GleeMessage extends EventEmitter {
   private _outbound: boolean
   private _cluster: boolean
   private _params: { [key: string]: string }
+  timestamp: string
+  key: string | Buffer
```

Comment:
You have to get rid of these two lines. They don't belong here.

Reply:
but then in the index.ts file it gives error saying that key & timestamp are not defined in GleeMessage

Reply:
True but that was just a typo you made there. `key` and `timestamp` are not properties of the message object but of the headers. Left another suggestion below.